objects/src/accounts/account_id.rs

```diff
@@ -792,6 +768,65 @@ pub(super) fn compute_digest(
     Hasher::hash_elements(&elements)
 }
 
+/// Creates an [`AccountId`] for testing purposes using the provided rng to sample randomness.
+///
+/// This uses [`AccountId::dummy`] under the hood, so refer to its documentation for details.
+///
+/// When only an rng is provided, the macro will generate a random [`AccountType`] and
+/// [`AccountStorageMode`] and pass it to [`AccountId::dummy`].
+///
+/// If an rng, an account type and a storage mode is given, only the random input to
+/// [`AccountId::dummy`] is randomly generated and the resulting ID will have the given account
+/// type and storage mode.
+///
+/// # Examples
+///
+/// ```
+/// # use crate::account_id_dummy;
+/// # use crate::accounts::{AccountType, AccountStorageMode};
+///
+/// let mut rng = rand::thread_rng();
+/// let random_id1 = account_id_dummy!(&mut rng);
+/// let random_id2 =
+///     account_id_dummy!(&mut rng, AccountType::FungibleFaucet, AccountStorageMode::Public);
+/// ```
+#[cfg(any(feature = "testing", test))]
+#[macro_export]
+macro_rules! account_id_dummy {
```

Comment:
nit (feel free to ignore): `dummy_account_id` feels a bit more natural as the name here.